### DISCUSSION TITLE
Library methods not using context

### DISCUSSION THREAD

**Discussion by gnuletik** [amacneil/dbmate#566] [Feature Ideas]
**Description**

Functions exposed by the library are not taking a `context.Context` in parameter.

This parameter allows cancelling the calls from the parent functions.

- Version: `v2.19.0`
- Database: All
- Operating System: All

**Steps To Reproduce**

See the public methods:
https://pkg.go.dev/github.com/amacneil/dbmate/v2/pkg/dbmate#DB

**Expected Behavior**

A context is expected to be used when interacting with a database.

**Comment by gena01** [comment]
Would be really nice to be able to use `context` with at least `Migrate()` and `Rollback()`. I am specifically looking to be able to pass in my own `context.Context` during migration and have it do `ExecuteContext(ctx, ..)` . This would let me handle timeouts, cancellations.

PS. `clickhouse-go` driver has a ton of options, parameters, etc that can be set via `context`. Parameter insertion is very interesting as it would let me extract things like passwords and such out of my migration scripts and maybe even store them elsewhere. This could also potentially be used to inject environment variables and have those be used in migration SQL.
